inlong-sort/sort-connectors/cdc-base/src/main/java/org/apache/inlong/sort/cdc/base/source/assigner/state/PendingSplitsStateSerializer.java

```diff
@@ -367,8 +370,9 @@ private List<TableId> readTableIds(DataInputDeserializer in) throws IOException
         List<TableId> tableIds = new ArrayList<>();
         final int size = in.readInt();
         for (int i = 0; i < size; i++) {
+            boolean useCatalogBeforeSchema = in.readBoolean();
             String tableIdStr = in.readUTF();
-            tableIds.add(TableId.parse(tableIdStr));
+            tableIds.add(TableId.parse(tableIdStr, useCatalogBeforeSchema));
```

Comment:
Please test if the task restore from savepoint is working properly.

Reply:
You can implement  `PostgresIncrementalSourceStreamFetcher` class by inheriting from `IncrementalSourceStreamFetcher` instead of modifying the logic in `IncrementalSourceStreamFetcher`. You can refer to https://github.com/apache/inlong/blob/master/inlong-sort/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/cdc/mongodb/source/reader/fetch/MongoDBScanFetchTask.java.